backend/src/app/util/pointer_map.clj

```diff
@@ -0,0 +1,224 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.util.pointer-map
+  (:require
+   [app.common.logging :as l]
+   [app.common.transit :as t]
+   [app.common.uuid :as uuid]
+   [app.util.fressian :as fres]
+   [app.util.time :as dt]
+   [clojure.core :as c])
+  (:import
+   clojure.lang.Counted
+   clojure.lang.IDeref
+   clojure.lang.IHashEq
+   clojure.lang.IObj
+   clojure.lang.IPersistentCollection
+   clojure.lang.IPersistentMap
+   clojure.lang.PersistentArrayMap
+   clojure.lang.PersistentHashMap
+   clojure.lang.Seqable
+   java.util.List))
+
+(def ^:dynamic *load-fn* nil)
+(def ^:dynamic *tracked* nil)
+(def ^:dynamic *metadata* {})
+
+(declare create)
+
+(defprotocol IPointerMap
+  (get-id [_])
+  (load! [_])
+  (modified? [_])
+  (clone [_]))
+
+(deftype PointerMap [id mdata
```

Comment:
I miss some documentation here. For those kind of structures that are not obvious, it's important to have an introduction of what is, more or less what is used for, and a brief general description of how does it work. All of this is complicated to guess without reviewing deeply the code itself.

This doesn't need to be very long, just a couple of paragraphs, perhaps, but it's important.

Reply:
:+1:

Reply:
done